dms/models/directory.py

```diff
@@ -661,17 +661,27 @@ def create(self, vals_list):
         return res
 
     def write(self, vals):
-        if vals.get("storage_id"):
+        if any([k in vals.keys() for k in ["storage_id", "parent_id"]]):
             for item in self:
-                if item.storage_id.id != vals["storage_id"]:
-                    raise UserError(_("It is not possible to change the storage."))
-        if vals.get("parent_id"):
-            parent = self.browse([vals["parent_id"]])
-            for item in self:
-                if item.parent_id.storage_id != parent.storage_id:
-                    raise UserError(
-                        _("It is not possible to change parent to other storage.")
-                    )
+                new_storage_id = (
+                    vals["storage_id"]
+                    if "storage_id" in vals.keys()
+                    else item.storage_id.id
+                )
+                new_parent_id = (
+                    vals["parent_id"]
+                    if "parent_id" in vals.keys()
+                    else item.parent_id.id
+                )
+                old_storage_id = (item.storage_id or item.parent_id.storage_id).id
```

Comment:
What happens if the directory is a third, fourth or so in depth level? I think there's a root_directory_id or so that should be used.